test/Datasource.test.js

```diff
@@ -57,10 +62,55 @@ describe('Test all datasouses from ../src/lib/datasouce', () => {
         });
     
         it('should resolve with null', ()=> {
-          return dataSource.delete().then((res) => {
-            expect(res).toEqual(null);  
+          return dataSource.delete('key').then((res) => {
+            expect(res).toEqual(true);  
           });
         });
       })
 
+      describe('#chaining', () => {
+        it('get shoud return null', () => {
+          return emptyDataSource.get('key').then((res) => {
+            expect(res).toEqual(null);  
+          });
+        });
+
+        it('set>get shoud return with seted value', () => {
+          return emptyDataSource.set('key', 'value')
+            .then(() =>  emptyDataSource.get('key').then())
+            .then((res) => {
+              expect(res).toEqual('value'); 
+            });
+        });
+
+        it('set>get>delete shoud return with true', () => {
+          return emptyDataSource.set('key', 'value')
+            .then(() =>  emptyDataSource.get('key').then())
+            .then(() =>  emptyDataSource.delete('key').then())
+            .then((res) => {
+              expect(res).toEqual(true); 
+            });
+        });
+
+        it('set>get>delete>get shoud return with null', () => {
+          return emptyDataSource.set('key', 'value')
+            .then(() =>  emptyDataSource.get('key').then())
+            .then(() =>  emptyDataSource.delete('key').then())
+            .then(() =>  emptyDataSource.get('key').then())
+            .then((res) => {
+              expect(res).toEqual(null); 
+            });
+        });
+
+        it('set>get>set>get shoud return with second seted value', () => {
+          return emptyDataSource.set('key', 'value')
+            .then(() =>  emptyDataSource.get('key').then())
+            .then(() =>  emptyDataSource.set('key', 'value1').then())
+            .then(() =>  emptyDataSource.get('key').then())
+            .then((res) => {
+              expect(res).toEqual('value1'); 
+              return emptyDataSource.delete('key').then()
```

Comment:
This .delete() method need for clear seted value, it is ok?

Reply:
Think it's easier to just create new dataSource in beforeEach() hook.
```
let datasource = null;
beforeEach() { datasource = new Datasource(); } 
```

Reply:
Ohh, this hook will calling after each test?

Reply:
beforeEach actually